components/menu/submenu-title.component.ts

```diff
@@ -96,12 +97,13 @@ export class NzSubMenuTitleComponent implements OnDestroy, OnInit {
   }
 
   setMouseState(state: boolean): void {
-    if (!this.nzDisabled) {
+    if ((!this.nzDisabled && this.nzTriggerSubMenuAction === 'hover') || !state) {
```

Comment:
`|| !state`  -> mouseLeave event will trigger closing the submenu

Reply:
done